CONTRIBUTING.md

```diff
@@ -1,124 +1,165 @@
+Alternatively, you can run the tests using `tox`, which is typically slower but more resilient, as it corresponds more precisely to way tests are run on GitHub before pull requests can be merged. To enable `tox`, ensure that it is installed:
 
-For the API reference, we document Python code inline, using [reST markup](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html). See [here](docs/README.md) for details on the documentation build. All parts of the public API need docstrings (indeed anything without docstrings won't appear in the built documentation). Similarly, don't add docstrings to private functionality, else it will appear in the documentation website. Use code comments sparingly, as they incur a maintenance cost and tend to drift out of sync with the corresponding code.
+````bash
+pip install tox
+````
 
+To run the notebooks, there are additional setup instructions described [here](docs/README.md).
 
 ### Quality checks
 
-We use [tox](https://tox.readthedocs.io) to run reproducible quality checks. This guide assumes you have this installed.
+[GitHub actions](https://docs.github.com/en/actions) will automatically run a number of quality checks against all pull requests to the develop branch. The GitHub repository is set up such that these need to pass in order to merge. Below is a summary of the various checks. We recommend manually running them before making a pull request, though it is also possible to wait until they are run on GitHub and fixing any errors in subsequent branch commits.
 
 #### Type checking
 
-We use [type hints](https://docs.python.org/3/library/typing.html) for documentation and static type checking with [mypy](http://mypy-lang.org). We do this throughout the source code and tests. The notebooks are checked for type correctness, but we only add types there if they are required for mypy to pass. This is because we anticipate most readers of notebooks won't be using type hints themselves. If you don't know how to add type hints when making a pull request, ask for help from reviewers or on the [community Slack workspace](./README.md#getting-help). You can use `typing.Any` where the actual type isn't expressible or practical, but do avoid it where possible.
+We use [type hints](https://docs.python.org/3/library/typing.html) for documentation and static type checking with [mypy](http://mypy-lang.org). We do this throughout the source code and tests. The notebooks are checked for type correctness, but we only add types there if they are required for mypy to pass. This is because we anticipate most readers of notebooks won't be using type hints themselves. If you don't know how to add type hints when making a pull request, ask for help from reviewers or on the [community Slack workspace](./README.md#getting-help).
 
 Run the type checker with
 ```bash
 $ tox -e types
 ```
 
+or, if running inside a virtual environment, just
+```bash
+mypy
```

Comment:
not `task mypy`?

Reply:
I have never tried it, but can you run it for a specific file, same as for tests?

Reply:
`task mypy` just runs `mypy` at the moment, but I'll change it in case we ever change the default beahviour. And yes, you can pass in an argument; I'll mention that.